### PR TITLE
fix(ng) Some API/User select related fixes

### LINKED ISSUE #1278: ng api-select - scroll bottom continues to call API after last page
Steps to reproduce:
- scroll down to the end of an API select (end = no more results, let's say page 5)
- scroll up a little
- scroll down
- a call is made to page 6
- scroll up a little
- scroll down
- a call is made to page 7
- ...

### LINKED ISSUE #1279: ng api-select - double HTTP call on page reset
Steps to reproduce:
- scroll to page 2 in an api-select
- update clue to "dan":
   - one call is made to `/api/...?search=dan&page=2`
   - one call is made to `/api/...?search=dan&page=1`

(Page is reset [here](https://github.com/LuccaSA/lucca-front/blob/293db7669f3b87cff825c58835ba6f249ef810bd/packages/ng/libraries/api/src/lib/select/searcher/api-searcher.component.ts#L131))

### LINKED ISSUE #1281: ng api-select - missing debounce
ApiPagedSearcher does not have a debounce anymore (it is commented out [here](https://github.com/LuccaSA/lucca-front/blob/293db7669f3b87cff825c58835ba6f249ef810bd/packages/ng/libraries/api/src/lib/select/searcher/api-searcher.component.ts#L132)).

Bonus: we could have a way to override the 250ms debounce as it is quite too fast

### LINKED ISSUE #1282: ng user-select - missing page reset on clue update
Steps to reproduce:
- scroll to page 3 inside a user-select
- search for "dan": API call still have `page=3`

(probably something around [here](https://github.com/LuccaSA/lucca-front/blob/293db7669f3b87cff825c58835ba6f249ef810bd/packages/ng/libraries/user/src/lib/select/searcher/user-searcher.component.ts#L93))

### PR DESCRIPTION
**Pull request by GuillaumeNury**
Can be read commit by commit:
* add missing debounce + add input for customization (fix #1281)
* stop calling API after last page (fix #1278)
* rework paging to avoid double calls (fix #1279) and "reset" when clue change (fix #1282)
* add storybook stories (should I keep them?)
* fix double subscribe to date-picker events (no ticket)

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1306

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1306

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1306

File: stories/ng/date/select/date-select.stories.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	]
+} as Meta;
+
+const template: Story<DateSelectStory> = (args: DateSelectStory) => ({
+	props: args,
+});
+
+export const basic = template.bind({});
+basic.args = {
+	model: "2021-05-06T12:00:00",
```
Comment (lucienbertin) [inline stories/ng/date/select/date-select.stories.ts:48]:
```suggestion
	model: "2021-05-06",
```

**Comment by LuccaIntegration** [comment]
jenkins auto deploy http://lucca-front.lucca.local/PR-1306

**Review by lucienbertin — APPROVED** [review]

**Pull request merged**